sky/skypilot_config.py

```diff
@@ -414,10 +415,26 @@ def override_skypilot_config(
         yield
         return
     original_config = _dict
+    override_configs = config_utils.Config(override_configs)
+    disallowed_diff_keys = []
+    for key in constants.SKIPPED_CLIENT_OVERRIDE_KEYS:
+        value = override_configs.pop_nested(key, default_value=None)
+        if (value is not None and
+                value != original_config.get_nested(key, default_value=None)):
+            disallowed_diff_keys.append(key)
```

Comment:
```suggestion
            disallowed_diff_keys.append('.'.join(key))
```
Apparently each key is a tuple. This should work for cases where the skipped key is nested, like you have on the test.
```
SKIPPED_CLIENT_OVERRIDE_KEYS: List[Tuple[str, ...]] = [('admin_policy',),
                                                       ('api_server',),
                                                       ('allowed_clouds',)]
```